org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/popup/AbstractNotificationPopup.java

```diff
@@ -0,0 +1,567 @@
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jface.resource.JFaceResources;
+import org.eclipse.jface.resource.LocalResourceManager;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ControlAdapter;
+import org.eclipse.swt.events.ControlEvent;
+import org.eclipse.swt.events.MouseAdapter;
+import org.eclipse.swt.events.MouseEvent;
+import org.eclipse.swt.events.MouseTrackAdapter;
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.GC;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.Point;
+import org.eclipse.swt.graphics.Rectangle;
+import org.eclipse.swt.graphics.Region;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Monitor;
+import org.eclipse.swt.widgets.Shell;
+import org.sonarlint.eclipse.ui.internal.SonarLintImages;
+import org.sonarlint.eclipse.ui.internal.popup.AnimationUtil.FadeJob;
+import org.sonarlint.eclipse.ui.internal.popup.AnimationUtil.IFadeListener;
+
+/**
+ * A popup window with a title bar and message area for displaying notifications.
+ * 
+ * Copied from Mylyn waiting for an inclusion in standard Eclipse
+ */
+public abstract class AbstractNotificationPopup extends Window {
+
+  private static final int TITLE_HEIGHT = 24;
+
+  private static final String LABEL_JOB_CLOSE = "Close Notification Job";
+
+  private static final int MAX_WIDTH = 400;
+
+  private static final int MIN_HEIGHT = 100;
+
+  private static final long DEFAULT_DELAY_CLOSE = 8 * 1000;
+
+  private static final int PADDING_EDGE = 5;
+
+  private long delayClose = DEFAULT_DELAY_CLOSE;
+
+  protected LocalResourceManager resources;
+
+  private GradientColors color;
+
+  private final Display display;
+
+  private Shell shell;
+
+  private Region lastUsedRegion;
+
+  private Image lastUsedBgImage;
+
+  private final Job closeJob = new Job(LABEL_JOB_CLOSE) {
+
+    @Override
+    protected IStatus run(IProgressMonitor monitor) {
+      if (!display.isDisposed()) {
+        display.asyncExec(new Runnable() {
+          public void run() {
+            Shell shell = AbstractNotificationPopup.this.getShell();
+            if (shell == null || shell.isDisposed()) {
+              return;
+            }
+
+            if (isMouseOver(shell)) {
+              scheduleAutoClose();
+              return;
+            }
+
+            AbstractNotificationPopup.this.closeFade();
+          }
+
+        });
+      }
+      if (monitor.isCanceled()) {
+        return Status.CANCEL_STATUS;
+      }
+
+      return Status.OK_STATUS;
+    }
+  };
+
+  private final boolean respectDisplayBounds = true;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this final field static too. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1170)